recipes/protobuf/all/conanfile.py

```diff
@@ -141,23 +141,6 @@ def _patch_sources(self):
             protoc_target
         )
 
-        # Set DYLD_LIBRARY_PATH in command line to avoid issues with shared protobuf
-        # (even with virtualrunenv, this fix might be required due to SIP)
-        # Only works with cmake, cmake_find_package or cmake_find_package_multi generators
-        if tools.is_apple_os(self.settings.os):
-            tools.replace_in_file(
-                protobuf_config_cmake,
-                "add_custom_command(",
-                ("set(CUSTOM_DYLD_LIBRARY_PATH ${CONAN_LIB_DIRS} ${Protobuf_LIB_DIRS} ${Protobuf_LIB_DIRS_RELEASE} ${Protobuf_LIB_DIRS_DEBUG} ${Protobuf_LIB_DIRS_RELWITHDEBINFO} ${Protobuf_LIB_DIRS_MINSIZEREL})\n"
-                 "string(REPLACE \";\" \":\" CUSTOM_DYLD_LIBRARY_PATH \"${CUSTOM_DYLD_LIBRARY_PATH}\")\n"
-                 "add_custom_command(")
-            )
-            tools.replace_in_file(
-                protobuf_config_cmake,
-                "COMMAND  protobuf::protoc",
-                "COMMAND ${CMAKE_COMMAND} -E env \"DYLD_LIBRARY_PATH=${CUSTOM_DYLD_LIBRARY_PATH}\" $<TARGET_FILE:protobuf::protoc>"
-            )
```

Comment:
I've tested without this block on Macos. I have CMake 3.22.0, and test package fails if shared. Injecting environment variables before calling CMake doesn't help.

Reply:
interesting. for me, it works for shared without this block. let's see what CI says...

Reply:
Do you have SIP enabled?

Reply:
yes, I do

Reply:
not sure what happens, but CI agrees with you, but locally for me it passes. I'll take a closer look tomorrow. maybe it's the fact I am using zsh shell from brew (which is not SIP protected).